### ISSUE TITLE
Proposal: MapError extensions

### ISSUE THREAD

**Issue by hankovich** [vkhorikov/CSharpFunctionalExtensions#210]
For now it's very easy to change `T` type (`Result<T, E>` to `Result<T2, E>`) with help of `Map` extensions but there's no ability to change `E` type.

Sometimes `Result<T>` should be converted to `Result<T, E>` or `Result<T, E>` should be converted to `Result<T, IEnumerable<T>>`.

Basically I'm talking about:
```csharp
Result<T, E> MapError<T, E>(this Result<T> result, Func<string, E> errorFactory);
Result<T, TNewError> MapError<T, E, TNewError>(this Result<T, E> result, Func<E, TNewError> errorFactory);

Task<Result<T, E>> MapError<T, E>(this Task<Result<T>> task, Func<string, E> errorFactory);
Task<Result<T, TNewError>> MapError<T, E, TNewError>(this Task<Result<T, E>> task, Func<E, TNewError> errorFactory);
```

And also probably:
```csharp
Result MapError(this Result result, Func<string, string> errorFactory);

Task<Result> MapError(this Task<Result> task, Func<string, string> errorFactory);
```

**Comment by vkhorikov** [comment]
Yep, that'd be helpful.

**Comment by benprime** [comment]
@hankovich Are you planning to submit a pull request for this one?

**Comment by hankovich** [comment]
@benprime thanks for pinging, totally forgot about this proposal.
Sure, will open a PR soon (till the end of this week)

**Comment by vkhorikov** [comment]
Merged, thank you! Will be available in 2.13.0.
